### DISCUSSION TITLE
玩遊戲也能學程式設計

### DISCUSSION THREAD

**Discussion by yt7l0** [1101-GS4719-C/Python-Community-GS4719-C#294] [閒聊八卦]
偶然發現一款能學程式設計的遊戲，JavaScript看起來是很難懂的程式語言，透過遊戲應該是很好入手的方式
[steam連結](https://store.steampowered.com/app/1812820/Bitburner/l)

**Comment by zixunn** [comment]
雖然多了一個遊戲的誘惑，但感覺有些人會因為玩個遊戲還要學JavaScript程式語言而選擇放棄。

**Reply by 109802503gloria** [reply]
我覺得會去玩的人應該就是有想學Javascript吧XD

**Comment by Isongzhe** [comment]
巴哈底下對這消息的留言都超好笑，什麼我上班都在打code ，下班還要繼續加班。不過感覺透過遊戲是真的讓學習門檻降低了

**Comment by Joan0518** [comment]
加上遊戲的誘惑還真的沒有吸引到我
光是學Python就讓我頭好痛

**Comment by 107602505** [comment]
學程式語言可以透過很多種方式學習，像是遊戲就是很好且無壓力的學習方式，另外像是由作業來自己找出需要的指令，在有點壓力的情境下我認為也不錯，因為寫出來的那一刻會很有成就感而且會漸漸熟悉寫程式語言的邏輯。

**Comment by YuHengChen54** [comment]
我覺得用遊戲學感覺成效不會很大，但就是在學習得當下會多一點樂趣

**Comment by JunTingLin** [comment]
上面說到降低寫程式的門檻: 這邊推薦一個google開發的[mblcok](https://blockly.games/)小遊戲🚩，完成後也會生成JavaScript，告訴你剛剛的每一步驟對應到的語法

**Comment by Rosancl** [comment]
感覺對於程式學習本身不會真的到非常有幫助，但是是一個滿棒的方式讓人願意開始，或者想要耍廢玩個遊戲但又不想真的那麼廢的時候就會是個好選擇哈哈哈

**Comment by Lai27330** [comment]
感覺是很有創意的點子，相較於購買厚重的原文書，遊戲或許具有更高的吸引力，但對於初學者來說可能觸動力仍然不及具有一定背景的使用者。

**Comment by pekoranoinu** [comment]
玩遊戲還能學程式，這真是很吸引人。

**Comment by lx3211** [comment]
感覺超讚的耶
拿來休閒多了解一個語言也不錯

**Comment by Jay-722** [comment]
雖然我玩遊戲絕對不會想玩這種看起來很燒腦的，但如果想要以輕鬆一點的方式來學習的話，可以嘗試看看。

**Comment by chupochun** [comment]
玩遊戲還能學程式語言，感覺真有趣

**Comment by 11timyyy** [comment]
玩遊戲又能增進腦力，真的還蠻吸引人!

**Comment by xkllkx** [comment]
我覺得這個youtube頻道在學習[java swift](https://www.youtube.com/playlist?list=PL-g0fdC5RMbpqZ0bmvJTgVTS4tS3txRVp)跟[python](https://www.youtube.com/playlist?list=PL-g0fdC5RMboYEyt6QS2iLb_1m7QcgfHk)很有幫助

**Comment by PoooYen** [comment]
這應該是讓基礎功力深厚的人在玩的吧XD 介面怎麼這麼像駭客任務

**Comment by ghost** [comment]
透過遊戲設計進入程式設計領域是挺不錯的，以這堂課來說，也能以pygame結合py來進修，個人嘗試練習過一次，說真的難店在於，類與類之間要如何分配負責的東西是需要思考琢磨的

**Comment by cordydeng** [comment]
這款遊戲我有下載來玩過，相較於拉方塊的程式設計遊戲，這款遊戲的介面比較偏向打程式時的環境，個人覺得對於完全新手的程式小白來說不太有吸引力

**Comment by Yuhsin000** [comment]
書跟遊戲二選一，當然用遊戲學比較好玩，之後試試看

**Comment by ThomasMeng0515** [comment]
對於想要學習寫程式概念的新手很適合!

**Comment by 110409538** [comment]
期待不只是寫程式可以遊戲化.

**Comment by HaoFish123** [comment]
感謝分享，但玩遊戲的人通常都是想放鬆吧。

**Comment by minniewei** [comment]
真的非常期待這些東西的產生！！！

**Comment by treeann** [comment]
真是一舉兩得呢

**Comment by stan890425** [comment]
這可以加入我收集遊戲的清單~~
果然steam是一個收集遊戲的遊戲

**Comment by jcpan821** [comment]
感謝分享~~

**Comment by alexthebigpp** [comment]
javescript是另一個很重要的程式語言以後可以用這個來學。

**Comment by Ben6uo** [comment]
steam上面有一個遊戲是用程式來控制的  'bitburner'

**Comment by bloodtrail0511** [comment]
感謝分享

**Comment by etin2468** [comment]
寫網頁很重要的語言，不知道這個遊戲實際效益有多少，感覺埋有趣的

**Comment by Lynn12378** [comment]
玩遊戲寫程式感覺能提供很大的吸引力呢
